run.ts

```diff
@@ -0,0 +1,258 @@
+/*
+ Copyright 2025 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import fs from "node:fs";
+import path from "node:path";
+import { execSync } from "node:child_process";
+
+type Vars = { [k: string]: string };
+
+type CISetup = {
+  env: Vars;
+  secrets: Vars;
+};
+
+type Config = {
+  "package-file": string[];
+  "ci-setup-filename": string;
+  "ci-setup-defaults": CISetup;
+  "ci-setup-help-url": string;
+  match: string[];
+  ignore: string[];
+  "exclude-packages": string[];
+};
+
+function usage(flags: string): string {
+  return [`usage: node run.ts ${flags}`].join("\n");
+}
+
+// The Makefile use .ONESHELL, which requires make 3.82 or higher.
+//    https://stackoverflow.com/a/32153249
+let make = "";
+function checkVersion(cmd: string, major: number, minor: number): boolean {
+  try {
+    const output = execSync(cmd).toString();
+    process.stdout.write(`${cmd} ${output.split("\n")[0]} -- `);
+    // console.log(`${cmd} ${output.split("\n")[0]}`);
+    const match = /(\d+)\.(\d+)/.exec(output);
+    if (match) {
+      const cmdMajor = parseInt(match[1]);
+      const cmdMinor = parseInt(match[2]);
+      if (cmdMajor > major || (cmdMajor === major && cmdMinor >= minor)) {
+        console.log(
+          `${cmdMajor}.${cmdMinor} is compatible with ${major}.${minor}`
+        );
+        return true;
+      }
+    }
+  } catch (_) {}
+  console.log(`not compatible with ${major}.${minor}`);
+  return false;
+}
+if (checkVersion("make --version", 3, 82)) {
+  make = "make";
+  console.log(`Using ${execSync("which make").toString().trim()}`);
+} else if (checkVersion("gmake --version", 3, 82)) {
+  make = "gmake";
+  console.log(`Using ${execSync("which gmake").toString().trim()}`);
+}
+
+if (!make) {
+  console.error("Error: make or gmake version 3.82 or higher is required.");
+  process.exit(1);
+}
+
+function lint(packagePath: string) {
+  const cmd = `${make} lint dir=${packagePath}`;
+  console.log(`>> ${cmd}`);
+  execSync(cmd, { stdio: "inherit" });
```

Comment:
Tried `node run.ts lint (path-to-code)`, and this failed with `gmake: *** No rule to make target 'lint'.  Stop.`. 

Tried this for nodejs-docs-samples/compute, and just nodejs-docs-samples. 

Unsure if the Makefile being run is the makefile in the cloud-samples-tools repo, or the one in the path.

Reply:
Even after fa0ddbf76815ef2c8efb4e37d1239f572a5f8800 this fails (that commit was for the test not lint). Comment stads

Reply:
Sorry for the lack of documentation. The script is meant to be run on the root directory of the langauge repo (not the tools repo) so for example:

`nodejs-docs-samples$ node run.ts test compute`

We still need to figure out and document the recommended way to get and run the script. We could just `curl <script-url> | node` to run it locally, and the GitHub action could checkout the specific file from the specific tag we want.